hepcrawl/parsers/elsevier.py

```diff
@@ -0,0 +1,578 @@
+                affiliations_by_id.append(", ".join(aff.extract()))
+            elif ce_affiliation:
+                aff = ce_affiliation.xpath(
+                    "./textfn/text()").extract_first()
+                aff = re.sub(r'^(\d+\ ?)', "", aff)
+                affiliations_by_id.append(aff)
+
+        return affiliations_by_id
+
+    def get_author_emails(self, author_node):
+        """Extract an author's email addresses."""
+        emails = author_node.xpath('./e-address[@type="email"]').extract()
+
+        return [remove_tags(email).strip() for email in emails]
+
+    @staticmethod
+    def get_author_name(author_node):
+        """Extract an author's name."""
+        surname = author_node.xpath('./surname').extract_first()
+        given_names = author_node.xpath('./given-name').extract_first()
+        suffix = author_node.xpath('.//suffix/text()').extract_first()
+        author_name = ', '.join(el for el in (surname, given_names, suffix) if el)
+
+        return remove_tags(author_name).strip()
+
+    @staticmethod
+    def get_root_node(elsevier_record):
+        """Get a selector on the root ``article`` node of the record.
+
+        This can be overridden in case some preprocessing needs to be done on
+        the XML.
+
+        Args:
+            elsevier_record(Union[str, scrapy.selector.Selector]): the record in Elsevier format.
+
+        Returns:
+            scrapy.selector.Selector: a selector on the root ``<article>``
+                node.
+        """
+        if isinstance(elsevier_record, six.string_types):
+            root = get_node(elsevier_record)
+        else:
+            root = elsevier_record
+        root.remove_namespaces()
+
+        return root
+
+    def get_author(self, author_node, author_group_node):
+        """Extract one author.
+
+        Args:
+            author_node(scrapy.selector.Selector): a selector on a single
+                author, e.g. a ``<contrib contrib-type="author">``.
+
+        Returns:
+            dict: the parsed author, conforming to the Inspire schema.
+        """
+        author_name = self.get_author_name(author_node)
+        emails = self.get_author_emails(author_node)
+        affiliations = self.get_author_affiliations(author_node, author_group_node)
+
+        return self.builder.make_author(author_name, raw_affiliations=affiliations, emails=emails)
+
+    @staticmethod
+    def get_reference_authors(ref_node):
+        """Extract authors from a reference node.
+
+        Args:
+            ref_node(scrapy.selector.Selector): a selector on a single reference.
+
+        Returns:
+            List[str]: list of names
+        """
+        given_names = ref_node.xpath(".//contribution/authors/author/given-name/text()").extract()
+        last_names = ref_node.xpath(".//contribution/authors/author/surname/text()").extract()
+        return [" ".join([name, last_name]) for name, last_name in zip(given_names, last_names)]
```

Comment:
are you sure you always have exactly one `given-name` and `surname` for each author? maybe it would be better to select `.//contribution/authors/author/`, then iterate on the nodes to extract given and surname for each of them.